TestApp/ContentView.swift

```diff
@@ -0,0 +1,184 @@
+                .font(.title2)
+
+            HStack{
+                Button("Privacy opt out"){
+                    MobileCore.setPrivacyStatus(PrivacyStatus.optedOut)
+                }.buttonStyle(CustomButton())
+
+                Button("Privacy opt in"){
+                    MobileCore.setPrivacyStatus(PrivacyStatus.optedIn)
+                }.buttonStyle(CustomButton())
+            }.padding()
+        }
+    }
+}
+
+struct NotificationSettingsCard: View {
+    @State var currentNotificationSettings = ""
+    @State var unreadNotificationCount = 0
+    @ObservedObject var pushDetail : PushNotificationDetailClass
+    var body: some View {
+        VStack {
+            Spacer(minLength: 20)
+            Text("Device Notification Settings")
+                .padding(.leading)
+                .font(.title2)
+
+            ScrollView{
+                Text(currentNotificationSettings).padding().font(.system(size: 13))
+            }.onAppear(perform: loadNotificationSettings)
+
+            Spacer(minLength: 20)
+            Text("Notification Details")
+                .padding([.leading,.bottom])
+                .font(.title2)
+            Text("Push Token: \(pushDetail.pushToken)").padding(.leading).font(.system(size: 13))
+            Text("Delivered Notifications: \(unreadNotificationCount)").padding().font(.system(size: 13))
+            Button("Refresh Notification Data") {
+                loadNotificationSettings()
+                readOtherNotificationDetails()
+            }.buttonStyle(CustomButton())
+        }
+    }
+    
+    func readOtherNotificationDetails() {
+        UNUserNotificationCenter.current().getDeliveredNotifications(completionHandler: { notifications in
+            unreadNotificationCount = notifications.count
+        })
+    }
+    
+    func loadNotificationSettings() {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            currentNotificationSettings = """
+            Authorization status \t \t: \(getAuthorizationStatusString(settings.authorizationStatus))
+            Notification Center Setting \t: \(getUNNotificationSettingString(settings.notificationCenterSetting))
+            Sound Setting \t \t \t: \(getUNNotificationSettingString(settings.soundSetting))
+            Badge Setting \t \t \t: \(getUNNotificationSettingString(settings.badgeSetting))
+            Alert Setting \t \t  \t \t: \(getUNNotificationSettingString(settings.alertSetting))
+            Announcement Setting \t: \(getUNNotificationSettingString(settings.announcementSetting))
+            LockScreen Setting \t \t: \(getUNNotificationSettingString(settings.lockScreenSetting))
+            Critical Alert Setting \t \t: \(getUNNotificationSettingString(settings.criticalAlertSetting))
+            TimeSensitive Setting \t \t: \(getUNNotificationSettingString(settings.timeSensitiveSetting))
+            """
+        }
+    }
+}
+
+struct CampaignClassicMockAPICard: View {
```

Comment:
nit: Im not sure why we need this when we have enabled the app for Push. We already have integration tests that help inspect and test the outgoing requests for the APIs (happy path).

Reply:
Most of the time for quick API testing we use the app with simulator. These mocks will certainly help on those times